### PR TITLE
SI-6117 regression involving import ambiguity. 

### PR DESCRIPTION
**Pull request by paulp**

### CODE REVIEW

**Comment by retronym** [comment]
I wonder, how did this ever work? Cursory inspection seems to absolve ddcb351.

**Comment by paulp** [comment]
retronym: Hey, you're right! But somehow it must be that commit.

**Comment by paulp** [comment]
retronym: Hey, it ISN'T that commit, I reverted it and it still fails.  Oddsmakers take a beating on that one.  Score one for cursory inspections.  Maybe it's martin and findMember changes? I don't know...

**Comment by paulp** [comment]
retronym: you knew to suspect yourself.  It's actually 666bbe730a, I haven't looked at how yet.

**Comment by paulp** [comment]
...and the shortest "how look" ever, I'm back.  Yeah.

**Comment by retronym** [comment]
The perils of a innocent looking `==`. I'm glad it was caught in time.

**Comment by paulp** [comment]
Since your spot and my spot are doing the identical check (and still differently for the moment) perhaps a little unification is in order.  I'm sort of glad this happened, it was worth it to discover the inequality of all overloaded symbols.

File: src/compiler/scala/tools/nsc/typechecker/Typers.scala [hunk 1]
```diff
@@ -4652,15 +4652,28 @@ trait Typers extends Modes with Adaptations with Tags {
...
+                  s"${impSym.debugLocationString} ${impSym.getClass}",
+                  s"${impSym1.debugLocationString} ${impSym1.getClass}"
+                ).mkString("\n  ")
+
+                // The symbol names are checked rather than the symbols themselves because
+                // each time an overloaded member is looked up it receives a new symbol.
+                // So foo.member("x") != foo.member("x") if x is overloaded.  This seems
+                // likely to be the cause of other bugs too...
+                if (t1 =:= t2 && impSym.name == impSym1.name)
+                  log(s"Suppressing ambiguous import: $t1 =:= $t2 && $impSym == $impSym1")
```
Comment (retronym) [inline src/compiler/scala/tools/nsc/typechecker/Typers.scala:4669]: `${impSym.name} == ${impSym1.name}`

(When do we get that tracer macro?)

**Comment by adriaanm** [comment]
(just dropping by to see how thing's are going here)

how are things going over here?

**Comment by retronym** [comment]
Paul's fix lgtm, other than the small fix needed to the logging output.

I guess the first cases could be collapsed into one, as Paul suggested, into something like:

``` scala
if (mt1 =:= mt2 && (
    (name.isTypeName && impSym.isMonomorphicType && impSym1.isMonomorphicType)) ||
     name.isTermName
))
```

But I'm not confident enough in my scala algebra to be sure, and it doesn't seem clearer that way.

**Comment by paulp** [comment]
I'd like to get it merged now if possible; I'll look at the modest cleanups.  I made the mistake of bundling the ant fixes with this, so meanwhile I continue to suffer with triple-recompiles on every change.

**Comment by adriaanm** [comment]
I'm fine merging it, the review status just wasn't clear to me.

**Pull request merged**